system/options.nix

```diff
@@ -0,0 +1,25 @@
+{ config, lib, ... }:
+let
+  inherit (lib.options) mkOption;
+  inherit (lib.types) str;
+in
+{
+  # More options can be added here.
+  # Following the module system, any module option added in the `options.aux` attribute set
+  # will become available under `config.aux` within your configuration. Thus making it
+  # possible to reference arbitrary internal variables if they have been set here
+  options.aux = {
+    hostname = mkOption {
```

Comment:
We should be extremely cautious about adding options. It doesn't seem like this one is used for anything other than mapping directly to networking.hostname. Maybe we can just use the available option and have an explicit "example" option if the intention is to show how to create options.

Reply:
This serves firstly to introduce the user to the module system, and secondly to avoid the user accidentally putting the hostname in specialArgs. Not exactly necessary, but not quite sure what else to put in a starter module system.